src/ChannelServer/World/Inventory.cs

```diff
@@ -232,12 +271,30 @@ public int AddSilent(Item item)
 		/// <return>Index of the item.</return>
 		public int Add(Item item, InventoryAddType addType)
 		{
-			var index = this.AddSilent(item);
+			Dictionary<int, int> indexes = this.AddSilent(item);
+
+			var lastItem = indexes.Last();
+			int lastIndex = lastItem.Key;
+
+			foreach (var itemObject in indexes)
+			{
+				int index = itemObject.Key;
+				int diff = itemObject.Value;
+				
+				if (index == lastIndex)
+				{
+					addType = InventoryAddType.PickUp;
```

Comment:
I'm not sure you understood what I meant, say I call `character.Inventory.Add(item, InventoryAddType.Something)` from somewhere. I specified that add type, because that, presumably, affects how the client displays the add, and I want to use `Something`. But you aren't using the value I pass to function, you override it.

What you should do is switch to NotNew if it actually is not new, or use what was passed to the function. Otherwise the parameter would be completely pointless and could be removed.